app/ocr.py

```diff
@@ -615,82 +556,100 @@ def get_outcome(self) -> List[str]:
 
         return outcomes_return
 
-    def get_state(self):
-        return "CA"
-        # """
-        # IF the case is an appellate decision the first page will show
-        # the legal teams' address on page 1, and the respondents address
-        # on page 2. We need to decide if appellate before getting state.
-        #
-        # Finds the city & state the respondent originally applied in. The function
-        # returns the state. City can be accessed as an attribute.
-        # """
-        #
-        # pattern = [
-        #     [{"LOWER": 'file'}],
-        #     [{"LOWER": 'files'}]
-        #     ]
-        #
-        # matches = similar(self, pattern)
-        #
-        # sentence = str(matches[0].sent)
-        # clean_sentence = sentence.replace(',', ' ').replace('\n', ' ')
-        #
-        # #if an appellate case the first page will have the location of the
-        # # appellate court, NOT origin
-        # if self.appellate:
-        #
-        #     #finds first state that's in the citycache
-        #     for abbrev in clean_sentence:
-        #         if abbrev in US_abbrev.keys():
-        #             return abbrev
-        #
-        # else:
-        #     #if non-appellate the first page will have the court location
-        #     # including city & state
-        #     for s in self.doc:
-        #         if s in US_abbrev.keys():
-        #             return s
+    def get_state(doc):
+        #Create list to hold matcher patterns from state dictionary
+        state_patterns = []
+        
+        for k in court_locs.keys():
+            state_patterns.append([{"LOWER": k.lower()}])
+        
+        #create matcher pattern for the key phrase file
+        file_pattern = [
+            [{"LOWER": 'file'}],
+            [{"LOWER": 'files'}]
+        ]
+        
+        #instantiate file pattern using the doc, this returns a span object that lists all the words pulled from the doc and their sentence and the index positions of the start of that sentence and end
+        file_matches = similar(doc, file_pattern) 
+        
+        #if spacy finds 'file' or 'files' we can grab the sentence that it contains. if not we can search the entire document for the state abbrev
+        if file_matches:
+            file_sentence = nlp(str(file_matches[0].sent))
+            #searches the sentence containing file for a state abbrev and returns the first one listed
+            state_matches = similar(file_sentence, state_patterns)
+            if state_matches:
+                return state_matches[0].text
+
+            #if file
+            else:
+                #if you change this output you need to change get_city to reflect that change.
+                return "Please select state"

```

Comment:
I'm not sure why your two else cases return different string to both essentially communicate "state not found".

Reply:
Jacob updated this as well, I did that for testing purposes and the "No Matches" should never trigger. You're correct though, this was changed.